fastlane/lib/fastlane/actions/create_keychain.rb

```diff
@@ -22,31 +24,31 @@ def self.run(params)
         end
 
         commands = []
-        commands << Fastlane::Actions.sh("security create-keychain -p #{escaped_password} #{keychain_path}", log: false)
+        commands << Fastlane::Actions.sh("security create-keychain -p #{escaped_password} #{keychain_path}", log: should_log)
 
         if params[:default_keychain]
           # if there is no default keychain - setting the original will fail - silent this error
           begin
-            Actions.lane_context[Actions::SharedValues::ORIGINAL_DEFAULT_KEYCHAIN] = Fastlane::Actions.sh("security default-keychain", log: false).strip
+            Actions.lane_context[Actions::SharedValues::ORIGINAL_DEFAULT_KEYCHAIN] = Fastlane::Actions.sh("security default-keychain", log: should_log).strip
           rescue
           end
-          commands << Fastlane::Actions.sh("security default-keychain -s #{keychain_path}", log: false)
+          commands << Fastlane::Actions.sh("security default-keychain -s #{keychain_path}", log: should_log)
         end
 
-        commands << Fastlane::Actions.sh("security unlock-keychain -p #{escaped_password} #{keychain_path}", log: false) if params[:unlock]
+        commands << Fastlane::Actions.sh("security unlock-keychain -p #{escaped_password} #{keychain_path}", log: should_log) if params[:unlock]
```

Comment:
I'd recommend using `Fastlane::Actions.sh_control_output(command, print_command: false, print_command_output: true)` here, just to always bias towards not logging the password.

Feel free to prepend that with a special call to `UI.command(command)` with the password parameter removed (when the  env var is present) if you want to get the rest of the command logged.